src/main/java/frc/robot/commands/DepositCommand.java

```diff
@@ -0,0 +1,125 @@
+package frc.robot.commands;
+
+import com.kauailabs.navx.frc.AHRS;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.commands.swerve.FollowPathCommand;
+import frc.robot.subsystems.drivetrain.BaseDrivetrain;
+import frc.robot.subsystems.drivetrain.BaseSwerveSubsystem;
+
+public class DepositCommand extends CommandBase {
+    private final BaseSwerveSubsystem driveSubsystem; // changed to BASE SWERVE SUBSYS to allow for getRobotPosition() 
+    private final AHRS ahrs; 
+
+    private final boolean red;
+
+
+    /*  all below setpoint constants are in inches
+        X axis --> long axis of field
+        Y axis --> short axis of field
+        origin at corner of blue grid 
+    */ 
+    private final double BLUE_X_SET = 86.0; 
+    private final double BLUE_Y_SET = 30.0;
+    private final double BLUE_R_SET = -90.0; // rotation in degrees
+    
+    private final double RED_X_SET = 564.0; 
+    private final double RED_Y_SET = 30.0;
+    private final double RED_R_SET = 90.0; // rotation in degrees
+
+    Pose2d blueSetPoint = new Pose2d(BLUE_X_SET, BLUE_Y_SET, Rotation2d.fromDegrees(BLUE_R_SET));
+    Pose2d redSetPoint = new Pose2d(RED_X_SET, RED_Y_SET, Rotation2d.fromDegrees(RED_R_SET));
+
+    private enum Stage{
+        Engaged,
+        OnAxis,
+        Normal2Node, 
+        Ready,
+        Deposited
+    }
+
+    private Stage stage;
```

Comment:
Instead of another multi-stage command, consider chaining several calls with `.andThen()` or extending `SequentialCommandGroup` and calling `addCommands()`? `FollowPathCommand` already waits until it is at the position and angle so there's no reason to add a third layer of checking. I'm not sure what `OnAxis` means, but `FollowPathCommand` also handles waypoints so your segmented path can be better expressed as a single `FollowPathCommand` that goes from the current position to the Y-axis Pose2d using a waypoint, or if the angle matters a chain of two `FollowPathCommand`s using the above command chaining methods. The `Normal2Node` stage doesn't seem necessary either, and `Ready` can be another command in the sequence. Maybe you can look at how `BaseAutonSequence` is implemented.